HerPublicWebsite.BusinessLogic/Services/RegularJobs/WorkingDayHelperService.cs

```diff
@@ -0,0 +1,47 @@
+using HerPublicWebsite.BusinessLogic.ExternalServices.Common;
+
+namespace HerPublicWebsite.BusinessLogic.Services.WorkingDayHelper;
+
+public interface IWorkingDayHelperService
+{
+    public Task<DateTime> AddWorkingDaysToDateTime(DateTime initialDateTime, int workingDaysToAdd);
+}
+
+public class WorkingDayHelperService : IWorkingDayHelperService
+{
+    public async Task<DateTime> AddWorkingDaysToDateTime(DateTime initialDateTime, int workingDaysToAdd)
+    {
+        var direction = workingDaysToAdd < 0 ? -1 : 1;
+        var holidays = await getHolidays();
+        var newDateTime = initialDateTime;
+        while (workingDaysToAdd != 0)
+        {
+            newDateTime = newDateTime.AddDays(direction);
+            if (newDateTime.DayOfWeek != DayOfWeek.Saturday && 
+                newDateTime.DayOfWeek != DayOfWeek.Sunday && 
+                !holidays.Contains(newDateTime))
```

Comment:
I think we rely on this being exactly at midnight - it might be safer to check `newDateTime.Date` here instead, so you can pass any `DateTime` to this method and it'll work correctly?

Reply:
Ah, good spot. Will update this. Thanks Joe!